File: flutter/craftd_widget/lib/data/model/action/analytics_properties.dart
class AnalyticsProperties {
  final String? category;
  final String? action;
  final String? label;
  final String? track;

  const AnalyticsProperties(
      {this.category, this.label, this.action, this.track});

  factory AnalyticsProperties.fromJson(dynamic properties) {
    return AnalyticsProperties(
      category: properties != null ? properties['category'] : null,
      action: properties != null ? properties['action'] : null,
      label: properties != null ? properties['label'] : null,
      track: properties != null ? properties['track'] : null,
    );
  }

  static String key = 'analytics';
}

File: android_kmp/craftd-core/src/commonMain/kotlin/com/github/codandotv/craftd/androidcore/data/model/action/AnalyticsProperties.kt
package com.github.codandotv.craftd.androidcore.data.model.action

import androidx.compose.runtime.Immutable
import androidx.compose.runtime.Stable
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Stable
@Immutable
@Serializable
data class AnalyticsProperties(
    @SerialName("category")
    val category: String? = null,

    @SerialName("action")
    val action: String? = null,

    @SerialName("label")
    val label: String? = null,

    @SerialName("track")
    val track: String? = null,
)
